packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -577,12 +630,19 @@ function completeWork(
         // TODO: Delete this when we delete isMounted and findDOMNode.
         workInProgress.effectTag &= ~Placement;
       }
-      updateHostContainer(workInProgress);
+      const rootContainerInstance = fiberRoot.containerInfo;
+      const rootHostContext = getRootHostContext(rootContainerInstance);
+      updateHostContainer(
+        workInProgress,
+        rootContainerInstance,
+        rootHostContext,
+      );
       break;
     }
     case HostComponent: {
-      popHostContext(workInProgress);
       const rootContainerInstance = getRootHostContainer();
+      const childHostContext = getHostContext();
```

Comment:
This seems fragile. We typically don't have to assume that a context needs to be passed in. How should I know when I can use context and not?

What prevents you from popping it later?

Reply:
The test I wrote :D

Reply:
I'm not interested in the bugs but why we can't restructure the code to also fix it.

Reply:
So the only alternative I could think of was to have expose a `getParentContext` function from the stack module. This seemed less bad? Idk